lib/node_modules/@stdlib/stats/base/dists/chisquare/mgf/src/main.c

```diff
@@ -0,0 +1,47 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/pow.h"
+
+/**
+* Evaluates the moment-generating function (MGF) for a chi-squared distribution
+* with degrees of freedom `k` at a value `t`.
+*
+* @param t    input value
+* @param k    degrees of freedom (must be non-negative)
+* @return     evaluated MGF, or NaN if input is invalid
+*
+* @example
+* double y = stdlib_base_dists_chisquare_mgf( 0.4, 2.0 );
+* // returns ~5.0
+*
+* @example
+* double y = stdlib_base_dists_chisquare_mgf( -1.0, 5.0 );
+* // returns ~0.0642
+*
+* @example
+* double y = stdlib_base_dists_chisquare_mgf( 0.0, 10.0 );
+* // returns 1.0
+*/
```

Comment:
Per convention, we only keep one example in the C files:

```suggestion
* @example
* double y = stdlib_base_dists_chisquare_mgf( 0.4, 2.0 );
* // returns ~5.0
*/
```